### PR TITLE
(MODULES-3133) Re-vendor DSC resources

### PR DESCRIPTION
**Pull request by Iristyle**

### CODE REVIEW

File: build/dsc/manager.rb [hunk 1]
```diff
@@ -214,6 +214,41 @@ def clean_dmtf_mofs
       clean_folder([@dmtf_mof_folder])
     end
 
+    def format_embeddedtype_all_values(property)
```
Comment (Iristyle) [inline build/dsc/manager.rb]: This isn't quite right as it fails when `manager.rb` has the extraneous types omitted.

**Comment by Iristyle** [comment]
@glennsarti pending another look by you, this should be good..

File: build/dsc/templates/dsc_type.rb.erb [hunk 2]
```diff
@@ -108,7 +108,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
           .select { |c| c[:klass].name == property.embeddedinstance_class_name }
           .first[:klass]
           .features
-          .map { |p| Dsc::Property.new(p) }
+          .map { |p| Dsc::Property.new(p, resource.relative_mof_path) }
```
Comment (ferventcoder) [inline build/dsc/templates/dsc_type.rb.erb]: Interesting.

File: build/dsc/resource.rb [hunk 3]
```diff
@@ -16,6 +16,7 @@ def initialize(mof_class, mof_path)
       @embedded_properties = nil
       @dsc_module          = nil
       @ps_module           = nil
+      @absentable          = nil
```
Comment (ferventcoder) [inline build/dsc/resource.rb]: Did I miss something? I thought we already had this...?

File: build/dsc/manager.rb [hunk 4]
```diff
@@ -32,9 +32,6 @@ def initialize
         'MSFT_Credential' => { 'user' => 'user', 'password' => 'password' },
         'MSFT_KeyValuePair' => { 'somekey' => 'somevalue' },
         'MSFT_KeyValuePair[]' => { 'somekey' => 'somevalue', 'somekey2' => 'somevalue2' },
-        # these hashes will be automatically wrapped in tests
-        'MSFT_xFileDirectory[]' => { 'DestinationPath' => 'c:/foo/bar', 'Recurse' => true },
-        'MSFT_xWebBindingInformation[]' => { 'Port' => 8080, 'Protocol' => 'https' },
```
Comment (ferventcoder) [inline build/dsc/manager.rb]: We were able to remove these?

**Comment by ferventcoder** [comment]
On the whole I'm +1 with what I can see. I imagine that the vendored files have changed as well.

File: build/dsc/resource.rb (hunk 3 above)
Comment (Iristyle) [inline build/dsc/resource.rb]: That's about setting a default so we can do a simple comparison to see if it's been calculated already (as we query `absentable` quite a bit when building templates)

File: build/dsc/manager.rb (hunk 4 above)
Comment (Iristyle) [inline build/dsc/manager.rb]: Last commits generates EmbeddedInstance spec values on the fly... no more need to statically define these.

**Comment by glennsarti** [comment]
@Iristyle Did you want to update the README_BUILD.md file with which DSC SHA reference you used? (0a9d28f) or leave that for MODULES-3179 to take care of.

File: build/dsc/templates/dsc_type.rb.erb (hunk 2 above)
Comment (glennsarti) [inline build/dsc/templates/dsc_type.rb.erb]: Without this being passed through, the Property can only emit error information with only the property name e.g. 'Property 'Ensure' is not valid'.  This is not useful during debugging as it does not tell you which of the 200+ MOF files have the property, particularly in our case which was debugging the Ensure property.  By passing the full path to the schema MOF file, useful error messages can be generated.

**Comment by Iristyle** [comment]
Thanks for catching that @glennsarti - updated `README` and repushed.

File: build/dsc/templates/dsc_type.rb.erb (hunk 2 above)
Comment (ferventcoder) [inline build/dsc/templates/dsc_type.rb.erb]: :+1:

File: build/dsc/manager.rb (hunk 4 above)
Comment (ferventcoder) [inline build/dsc/manager.rb]: :+1:

**Pull request merged**